client/src/main/java/io/pravega/client/netty/impl/ClientConnectionInboundHandler.java

```diff
@@ -128,7 +128,8 @@ public void sendAsync(List<Append> appends, CompletedCallback callback) {
         promise.addListener(new GenericFutureListener<Future<? super Void>>() {
             @Override
             public void operationComplete(Future<? super Void> future) throws Exception {
-                callback.complete(new ConnectionFailedException(future.cause()));
+                Throwable cause = future.cause();
```

Comment:
future is not an exception, so this is not getting an inner cause, its just getting the exception that was thrown or null if none was.